packages/transition-backend/src/services/simulation/methods/OdTripSimulation.ts

```diff
@@ -171,8 +173,71 @@ export default class OdTripSimulation implements SimulationMethod {
+        //this.wrappedJob.job.getReadStream('input')..
+        /*const inputFiles: {
+            [Property in keyof BatchRouteJobType[fileKey]]?:
+                | string
+                | { filepath: string; renameTo: string };
+        } = {};
+        inputFiles.input = await ExecutableJobUtils.prepareJobFiles(
+            parameters.fileAndMapping.csvFile,
+            userId
+            );*/
+
+
+
+        
+        // Need to build a BatchCalculationParameters for the BatchRouteJobType
+        // It's composed TransitRoutingQueryAttributes plus the withGeometries, detailed flag
+        // The TransitRoutingQueryAttributes is a RoutingQueryAttributes + TransitQueryAttributes
+        // The TransitQueryAttributes is a TransitRoutingBaseAttributes (coming from options.transitRoutingAttributes)
+        // and a scenarioId. The RoutingQueryAttributes is the routingModes and the withAlternatives flag. 
+        const batchParams: BatchCalculationParameters = {
+            ...options.transitRoutingAttributes,
+            routingModes: ['transit', 'walking', 'driving'], //We need walking and driving for the fallback calculation
+            withAlternatives: false,
+            withGeometries: false,
+            detailed: false,
+            scenarioId: _scenarioId
+        };
+
+        const routingJob: ExecutableJob<BatchRouteJobType> = this.wrappedJob.job.createChildJob({
+            user_id: this.wrappedJob.job.attributes.user_id,
+            name: 'batchRoute', //TODO Is this important, can I rename it do something else ???
+            data: {
+                parameters: {
+                    demandAttributes: this.wrappedJob.job.attributes.demandAttributes,
+                    transitRoutingAttributes: batchParams
+                }
+            },
+            inputFiles,
+            hasOutputFiles: true //TODO Manage the result
+        });
+
+        //TODO Normally we would yeild the execution here. To let the child run. For now run it directly.
+        // I would normally do routingJob.run() here, but it was not implemented like that :P
+
+        // This is copied from wrapBatchRoute
+        const { files, errors, warnings, ...result } = await batchRoute(routingJob, {
+            progressEmitter: newProgressEmitter(task), // TODO Figure out how to get EvolutionaryTransitNetworkDesignJobExecutor progressEmitter (If we want, we could just have a new one of a noop)
+            isCancelled: getTaskCancelledFct(task) // TODO Figure out how to get EvolutionaryTransitNetworkDesignJobExecutor isCancelled function
+        });
+        routingJob.attributes.data.results = result;
+        routingJob.attributes.resources = { files };
+
+        // TODO Get job results somehow
+        // TODO Guessing we could transform the processResults here in some kind of result visitor and figure out a way to pass it
```

Comment:
Your guess is correct, that is how I would do. The `batchRoute` could receive an optional visitor as parameter